### PR TITLE
DVC-6487 switch NodeJS SDK to use variableForUser()

### PR DESCRIPTION
**Pull request by jonathannorris**
- Will switch to using protobuf interface in separate PR, wanted to get this working first incase we run into complications with protobuf

### CODE REVIEW

File: sdk/nodejs/src/utils/userBucketingHelper.ts [hunk 1]
```diff
@@ -7,3 +7,26 @@ export function bucketUserForConfig(user: DVCPopulatedUser, token: string): Buck
         getBucketingLib().generateBucketedConfigForUser(token, JSON.stringify(user))
     ) as BucketedUserConfig
 }
+
+export function getVariableTypeCode(type: VariableType): number {
+    // TODO: can't figure out how to get the enum values from the bucketing lib as its dynamically imported
```
Comment (jonathannorris) [inline sdk/nodejs/src/utils/userBucketingHelper.ts:12]: @ajwootto any idea how to handle this? I tried loading the enum from the `Exports` of `@devcycle/bucketing-assembly-script` and from the `getBucketingLib()` object itself that is loaded with no luck.

Comment (jonathannorris) [reply sdk/nodejs/src/utils/userBucketingHelper.ts:12]: it gets compiled to this: 
image

**Comment by jonathannorris** [comment]
Current dependencies on/for this PR:
* main
  * **PR #435** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/435" target="_blank">Graphite</a>  👈
    * **PR #436** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/436" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/435?utm_source=stack-comment).

File: sdk/nodejs/__tests__/client.spec.ts [hunk 2]
```diff
@@ -90,58 +95,4 @@ describe('variable', () => {
         // this will be a type error for non-JSON
         console.log(variable.value.asdasdas)
     })
-
-    describe('variableDefaulted event', () => {
```
Comment (ajwootto) [inline sdk/nodejs/__tests__/client.spec.ts:94]: why delete?

File: sdk/nodejs/src/utils/userBucketingHelper.ts (hunk 1 above)
Comment (ajwootto) [reply sdk/nodejs/src/utils/userBucketingHelper.ts:12]: you can maybe look at how I did it in Go. They were available as globals named `VariableType.Boolean` etc.

File: sdk/nodejs/__tests__/client.spec.ts [hunk 3]
```diff
@@ -59,12 +60,16 @@ describe('variable', () => {
     })
 
     it('returns a valid variable object for a variable that is not in the config', () => {
+        // @ts-ignore
+        getBucketingLib().variableForUser.mockReturnValueOnce(null)
```
Comment (ajwootto) [inline sdk/nodejs/__tests__/client.spec.ts:64]: doesnt matter that much but you can also use the `jest.Mocked` type or whatever its called to give you correct typing here

**Review by ajwootto — APPROVED** [review]

**Pull request merged**